backend/app/services/auth_service.py

```
@@ -281,21 +283,108 @@ async def refresh_token(
+    settings: Settings = Depends(get_settings),
     request_id: Optional[str] = Depends(get_request_id),
-    _: bool = Depends(require_permissions([Permission.VIEW_OWN_PROFILE]))
+    _= Depends(require_permissions_dependency([Permission.VIEW_OWN_PROFILE]))
 ) -> dict:
-    """Retrieve profile information for the current user."""
+    """Retrieve profile information for the current user, including roles and permissions."""
     try:
+        # Validate SystemAction.PROFILE_UPDATE
+        if not await validate_enum_value(SystemAction, SystemAction.PROFILE_UPDATE.value):
+            logger.warning(f"Invalid SystemAction value: {SystemAction.PROFILE_UPDATE.value}")
+            raise ValidationError(detail=f"Invalid system action: {SystemAction.PROFILE_UPDATE.value}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)